hayeon/programmers-1/추억-점수.js

```diff
@@ -0,0 +1,32 @@
+function solution(name, yearning, photo) {
+  let answer = [];
+  const yearningChart = {};
+
+  name.forEach((el, i) => {
+    yearningChart[el] = yearning[i];
+  });
+
+  answer = photo.map((el) => {
+    let sumOfPoints = 0;
+
+    el.forEach((point) => {
+      if (yearningChart[point]) {
+        sumOfPoints += yearningChart[point];
+      } else {
+        sumOfPoints += 0;
+      }
+    });
+
+    return sumOfPoints;
+  });
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(name, yearning, photo) {
  const yearningChart = {};

  name.forEach((el, i) => {
    yearningChart[el] = yearning[i];
  });

  return photo.map((el) => {
    let sumOfPoints = 0;

    el.forEach((point) => {
      if (yearningChart[point]) {
        sumOfPoints += yearningChart[point];
      } else {
        sumOfPoints += 0;
      }
    });

    return sumOfPoints;
  });
}
```
- 풀이 잘봤습니다  ~!
- photo.map을 곧바로 리턴하면 answer는 불 필요한 변수입니다.
- 그리고 기존에도 const 변수인 yearingChart가 answer보다 상단에 위치하는게 좋습니다. (상수가 최상단)